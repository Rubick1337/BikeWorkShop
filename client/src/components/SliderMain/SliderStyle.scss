.container__slider {
  padding: 20px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 30px;
}

.title_section {
  display: flex;
  flex-direction: column;
  gap: 21px;
  justify-content: center;
  align-items: center;
}

.line__title {
  width: 170px;
  height: 2px;
  background: #0C0C0E;
}

.title_section h3 {
  color: #0C0C0E;
  font-family: Forum;
  font-size: 40px;
  font-weight: 400;
  line-height: 40px;
  letter-spacing: 2px;
  text-transform: capitalize;
}

.container__slider h4 {
  color: #000;
  text-align: center;
  font-family: Inter;
  font-size: 20px;
  font-weight: 400;
  line-height: 40px;
}

.slider__container {
  width: 1022px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  position: relative;
}

.slider__items {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
}

.image__item {
  height: 560px;
  background-size: cover;
  border-radius: 50px;
  background-position: center;
  background-repeat: no-repeat;
  width: 922px;
  transition: opacity 0.5s ease-in-out, visibility 0.5s ease-in-out;
  opacity: 0; /* Скрываем слайд по умолчанию */
  visibility: hidden; /* Прячем слайды */
  position: absolute; /* Размещаем слайды друг на друге */
}

.image__item.active {
  opacity: 1; /* Показываем активный слайд */
  visibility: visible; /* Делаем слайд видимым */
  position: relative; /* Даем активному слайду относительное позиционирование */
}

.pagination {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
}

.button {
  width: 26px;
  height: 26px;
  display: inline-block;
  background-color: transparent;
  cursor: pointer;
  border-radius: 50%;
}

.button .inner {
  width: 16px;
  height: 16px;
  margin: 5px;
  background-color: #0C0C0E;
  border-radius: 50%;
  transition: background-color 0.3s ease;
}

.inner:hover {
  background-color: #BB945F;
}

.inner:active {
  background-color: #BB945F;
}

.left-carrect, .right-carrect {
  width: 48px;
  height: 48px;
  background-size: cover;
  background-position: center;
  cursor: pointer;
  transition: transform 0.3s ease;
}

.left-carrect {
  background-image: url(../../images/Carret_Left.png);
}

.right-carrect {
  background-image: url(../../images/Carret_Right.png);
}

.left-carrect:hover,
.right-carrect:hover {
  transform: scale(1.1);
}

.button.active .inner {
  background-color: #BB945F;
}
@media (max-width: 1024px)
{
  .slider__container
  {
    width:730px;
  }
  .image__item
  {
    height: 400px;
    width: 730px;
  }
}
@media (max-width: 730px)
{
  .slider__container
  {
    width:530px;
  }
  .image__item
  {
    height: 400px;
    width: 530px;
  }
}
@media (max-width: 530px)
{
  .container__slider
  {
    display: none;
  }
}