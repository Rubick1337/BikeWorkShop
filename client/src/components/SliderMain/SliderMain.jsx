import React, { useState } from "react";
import "./SliderStyle.scss";


import image1 from '../../images/image.png';
import image2 from '../../images/frameLoading.png';
import image3 from '../../images/image.png';
import image4 from '../../images/image.png';
import image5 from '../../images/image.png';

export default function SliderMain() {
    const [currentSlide, setCurrentSlide] = useState(0);
    const images = [image1, image2, image3, image4, image5];

    // Функции для смены слайдов
    const goToSlide = (index) => {
        setCurrentSlide(index);
    };

    const nextSlide = () => {
        setCurrentSlide((prev) => (prev + 1) % images.length);
    };

    const prevSlide = () => {
        setCurrentSlide((prev) => (prev - 1 + images.length) % images.length);
    };

    return (
        <div className="container__slider">
            <div className="title_section">
                <h3>Bike workshop</h3>
                <div className="line__title"></div>
            </div>
            <h4>Веломастерская позаботиться о вашем велосипеде</h4>

            <div className="slider__container">
                <div className="slider__items">
                    <div className="left-carrect" onClick={prevSlide}></div>
                    {images.map((image, index) => (
                        <div
                            key={index}
                            className={`image__item ${currentSlide === index ? "active" : ""}`}
                            style={{ backgroundImage: `url(${image})` }}
                        ></div>
                    ))}
                    <div className="right-carrect" onClick={nextSlide}></div>
                </div>

                <div className="pagination">
                    {images.map((_, index) => (
                        <div
                            key={index}
                            className={`button ${currentSlide === index ? "active" : ""}`}
                            onClick={() => goToSlide(index)}
                        >
                            <div className="inner"></div>
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
}
