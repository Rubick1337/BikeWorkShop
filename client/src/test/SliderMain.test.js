import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import SliderMain from '../components/SliderMain/SliderMain'; // Убедитесь, что путь правильный

describe('SliderMain Component', () => {
    beforeEach(() => {
        render(<SliderMain />);
    });

    it('renders correctly with initial state', () => {
        // Проверяем, что заголовок отображается
        expect(screen.getByText(/Bike workshop/i)).toBeInTheDocument();
        expect(screen.getByText(/Веломастерская позаботиться о вашем велосипеде/i)).toBeInTheDocument();

        // Проверяем, что первый слайд активен
        const activeSlide = screen.getByRole('button', { name: /Go to slide 1/i });
        expect(activeSlide).toHaveClass('active');
    });

    it('changes slide when next button is clicked', () => {
        // Находим кнопку для следующего слайда
        const nextButton = screen.getByRole('button', { name: /Next slide/i });

        // Имитируем клик по кнопке "Next"
        fireEvent.click(nextButton);

        // Проверяем, что второй слайд активен
        const activeSlide = screen.getByRole('button', { name: /Go to slide 2/i });
        expect(activeSlide).toHaveClass('active');
    });

    it('changes slide when prev button is clicked', () => {
        // Находим кнопку для следующего слайда
        const nextButton = screen.getByRole('button', { name: /Next slide/i });
        fireEvent.click(nextButton); // Переходим на второй слайд

        // Находим кнопку для предыдущего слайда
        const prevButton = screen.getByRole('button', { name: /Previous slide/i });

        // Имитируем клик по кнопке "Prev"
        fireEvent.click(prevButton);

        // Проверяем, что первый слайд снова активен
        const activeSlide = screen.getByRole('button', { name: /Go to slide 1/i });
        expect(activeSlide).toHaveClass('active');
    });

    it('changes slide when pagination button is clicked', () => {
        // Находим кнопку пагинации для второго слайда
        const paginationButton = screen.getByRole('button', { name: /Go to slide 2/i });

        // Имитируем клик по кнопке пагинации
        fireEvent.click(paginationButton);

        // Проверяем, что второй слайд активен
        const activeSlide = screen.getByRole('button', { name: /Go to slide 2/i });
        expect(activeSlide).toHaveClass('active');
    });

    it('matches snapshot', () => {
        // Сравниваем снэпшот
        const { asFragment } = render(<SliderMain />);
        expect(asFragment()).toMatchSnapshot();
    });
});