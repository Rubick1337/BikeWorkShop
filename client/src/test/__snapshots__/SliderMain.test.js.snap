// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderMain Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container__slider"
  >
    <div
      class="title_section"
    >
      <h3>
        Bike workshop
      </h3>
      <div
        class="line__title"
      />
    </div>
    <h4>
      Веломастерская позаботиться о вашем велосипеде
    </h4>
    <div
      class="slider__container"
    >
      <div
        class="slider__items"
      >
        <div
          aria-label="Previous slide"
          class="left-carrect"
          role="button"
        />
        <div
          class="image__item active"
          style="background-image: url(image.png);"
        />
        <div
          class="image__item "
          style="background-image: url(bike.png);"
        />
        <div
          class="image__item "
          style="background-image: url(bike3.png);"
        />
        <div
          class="image__item "
          style="background-image: url(bike5.png);"
        />
        <div
          class="image__item "
          style="background-image: url(bike6.png);"
        />
        <div
          aria-label="Next slide"
          class="right-carrect"
          role="button"
        />
      </div>
      <div
        class="pagination"
      >
        <div
          aria-label="Go to slide 1"
          class="button active"
          role="button"
        >
          <div
            class="inner"
          />
        </div>
        <div
          aria-label="Go to slide 2"
          class="button "
          role="button"
        >
          <div
            class="inner"
          />
        </div>
        <div
          aria-label="Go to slide 3"
          class="button "
          role="button"
        >
          <div
            class="inner"
          />
        </div>
        <div
          aria-label="Go to slide 4"
          class="button "
          role="button"
        >
          <div
            class="inner"
          />
        </div>
        <div
          aria-label="Go to slide 5"
          class="button "
          role="button"
        >
          <div
            class="inner"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
